Build fails because of `Theme.AppCompat.Light.NoActionBar not foundà

### What
- Build fails because of `Theme.AppCompat.Light.NoActionBar not foundà
- Introduced by https://github.com/openfoodfacts/smooth-app/pull/981

### Log
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:/home/runner/work/smooth-app/smooth-app/packages/smooth_app/build/app/intermediates/packaged_manifests/debug/AndroidManifest.xml:62: AAPT: error: resource style/Theme.AppCompat.Light.NoActionBar (aka org.openfoodfacts.app:style/Theme.AppCompat.Light.NoActionBar) not found.

--- a/packages/smooth_app/android/app/src/main/AndroidManifest.xml
+++ b/packages/smooth_app/android/app/src/main/AndroidManifest.xml
@@ -47,10 +47,6 @@
                         <category android:name="android.intent.category.LAUNCHER"/>
                     </intent-filter>
                 </activity>
-                <activity
-                    android:name="com.yalantis.ucrop.UCropActivity"
-                    android:screenOrientation="portrait"
-                    android:theme="@style/Theme.AppCompat.Light.NoActionBar"/>
 
                 <!-- Don't delete the meta-data below.
                      This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->

--- a/packages/smooth_app/android/app/src/profile/AndroidManifest.xml
+++ b/packages/smooth_app/android/app/src/profile/AndroidManifest.xml
@@ -47,10 +47,6 @@
                 <category android:name="android.intent.category.LAUNCHER"/>
             </intent-filter>
         </activity>
-        <activity
-            android:name="com.yalantis.ucrop.UCropActivity"
-            android:screenOrientation="portrait"
-            android:theme="@style/Theme.AppCompat.Light.NoActionBar"/>
 
         <!-- Don't delete the meta-data below.
              This is used by the Flutter tool to generate GeneratedPluginRegistrant.java -->

--- a/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
+++ b/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
@@ -56,6 +56,7 @@ class _ImageUploadCardState extends State<ImageUploadCard> {
         imageField: widget.imageField,
         imageUri: croppedImageFile.uri,
       );
+      croppedImageFile.delete();
       if (isUploaded) {
         widget.onUpload(context);
       }

--- a/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
+++ b/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
@@ -4,6 +4,7 @@ import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:smooth_app/helpers/picture_capture_helper.dart';
+import 'package:smooth_app/pages/image_crop_page.dart';
 import 'package:smooth_app/pages/product/product_image_page.dart';
 
 class ImageUploadCard extends StatefulWidget {
@@ -33,7 +34,12 @@ class _ImageUploadCardState extends State<ImageUploadCard> {
       _imageFullProvider; // Full resolution image to display in image page
 
   Future<void> _getImage() async {
-    final File? croppedImageFile = await pickImageAndCrop(context);
+    final File? croppedImageFile = await Navigator.push<File?>(
+      context,
+      MaterialPageRoute<File?>(
+        builder: (BuildContext context) => ImageCropPage(),
+      ),
+    );
 
     if (croppedImageFile != null) {
       setState(() {

--- a/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
+++ b/packages/smooth_app/lib/cards/data_cards/image_upload_card.dart
@@ -33,7 +33,7 @@ class _ImageUploadCardState extends State<ImageUploadCard> {
       _imageFullProvider; // Full resolution image to display in image page
 
   Future<void> _getImage() async {
-    final File? croppedImageFile = await pickImageAndCrop();
+    final File? croppedImageFile = await pickImageAndCrop(context);
 
     if (croppedImageFile != null) {
       setState(() {

--- a/packages/smooth_app/lib/helpers/picture_capture_helper.dart
+++ b/packages/smooth_app/lib/helpers/picture_capture_helper.dart
@@ -30,14 +30,13 @@ Future<File?> pickImageAndCrop(BuildContext context) async {
     context,
     MaterialPageRoute<Widget>(
       builder: (BuildContext context) {
-        final ThemeProvider themeProvider = context.watch<ThemeProvider>();
+        context.watch<ThemeProvider>();
         final ThemeData theme = Theme.of(context);
         return Scaffold(
           body: Crop(
             image: bytes!,
             controller: _controller,
             onCropped: (Uint8List image) {
-              print('M123 $image');
               bytes = image;
               Navigator.pop(context);
             },

--- a/packages/smooth_app/lib/helpers/picture_capture_helper.dart
+++ b/packages/smooth_app/lib/helpers/picture_capture_helper.dart
@@ -1,14 +1,18 @@
 import 'dart:io';
+import 'dart:typed_data';
 
-import 'package:flutter/cupertino.dart';
+import 'package:crop_your_image/crop_your_image.dart';
+import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:image_cropper/image_cropper.dart';
 import 'package:image_picker/image_picker.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:provider/provider.dart';
 import 'package:smooth_app/database/product_query.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
 import 'package:smooth_app/widgets/loading_dialog.dart';
 
-Future<File?> pickImageAndCrop() async {
+Future<File?> pickImageAndCrop(BuildContext context) async {
   final ImagePicker picker = ImagePicker();
 
   final XFile? pickedXFile = await picker.pickImage(
@@ -19,13 +23,63 @@ Future<File?> pickImageAndCrop() async {
     return null;
   }
 
-  return ImageCropper.cropImage(
-    sourcePath: pickedXFile.path,
-    androidUiSettings: const AndroidUiSettings(
-      lockAspectRatio: false,
-      hideBottomControls: true,
+  Uint8List? bytes = await pickedXFile.readAsBytes();
+  final CropController _controller = CropController();
+
+  await Navigator.push<Widget>(
+    context,
+    MaterialPageRoute<Widget>(
+      builder: (BuildContext context) {
+        final ThemeProvider themeProvider = context.watch<ThemeProvider>();
+        final ThemeData theme = Theme.of(context);
+        return Scaffold(
+          body: Crop(
+            image: bytes!,
+            controller: _controller,
+            onCropped: (Uint8List image) {
+              print('M123 $image');
+              bytes = image;
+              Navigator.pop(context);
+            },
+            initialSize: 0.5,
+            baseColor: theme.colorScheme.primary,
+            maskColor: Colors.white.withAlpha(100),
+            cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
+                DotControl(color: theme.colorScheme.primary),
+          ),
+          bottomNavigationBar: BottomAppBar(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                IconButton(
+                  icon: const Icon(Icons.cancel_outlined),
+                  onPressed: () {
+                    bytes = null;
+                    Navigator.pop(context);
+                  },
+                ),
+                IconButton(
+                  icon: const Icon(Icons.done),
+                  onPressed: () {
+                    _controller.crop();
+                  },
+                )
+              ],
+            ),
+          ),
+        );
+      },
     ),
   );
+
+  if (bytes == null) {
+    return null;
+  }
+
+  final Directory tempDir = await getTemporaryDirectory();
+  final String tempPath = tempDir.path;
+  final String filePath = '$tempPath/upload_img_file_01.tmp';
+  return File(filePath).writeAsBytes(bytes!);
 }
 
 Future<bool> uploadCapturedPicture(

--- a/packages/smooth_app/lib/helpers/picture_capture_helper.dart
+++ b/packages/smooth_app/lib/helpers/picture_capture_helper.dart
@@ -1,85 +1,8 @@
-import 'dart:io';
-import 'dart:typed_data';
-
-import 'package:crop_your_image/crop_your_image.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:image_picker/image_picker.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
-import 'package:path_provider/path_provider.dart';
-import 'package:provider/provider.dart';
 import 'package:smooth_app/database/product_query.dart';
-import 'package:smooth_app/themes/theme_provider.dart';
 import 'package:smooth_app/widgets/loading_dialog.dart';
-
-Future<File?> pickImageAndCrop(BuildContext context) async {
-  final ImagePicker picker = ImagePicker();
-
-  final XFile? pickedXFile = await picker.pickImage(
-    source: ImageSource.camera,
-  );
-  if (pickedXFile == null) {
-    // User didn't pick any image.
-    return null;
-  }
-
-  Uint8List? bytes = await pickedXFile.readAsBytes();
-  final CropController _controller = CropController();
-
-  await Navigator.push<Widget>(
-    context,
-    MaterialPageRoute<Widget>(
-      builder: (BuildContext context) {
-        context.watch<ThemeProvider>();
-        final ThemeData theme = Theme.of(context);
-        return Scaffold(
-          body: Crop(
-            image: bytes!,
-            controller: _controller,
-            onCropped: (Uint8List image) {
-              bytes = image;
-              Navigator.pop(context);
-            },
-            initialSize: 0.5,
-            baseColor: theme.colorScheme.primary,
-            maskColor: Colors.white.withAlpha(100),
-            cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
-                DotControl(color: theme.colorScheme.primary),
-          ),
-          bottomNavigationBar: BottomAppBar(
-            child: Row(
-              mainAxisAlignment: MainAxisAlignment.spaceBetween,
-              children: <Widget>[
-                IconButton(
-                  icon: const Icon(Icons.cancel_outlined),
-                  onPressed: () {
-                    bytes = null;
-                    Navigator.pop(context);
-                  },
-                ),
-                IconButton(
-                  icon: const Icon(Icons.done),
-                  onPressed: () {
-                    _controller.crop();
-                  },
-                )
-              ],
-            ),
-          ),
-        );
-      },
-    ),
-  );
-
-  if (bytes == null) {
-    return null;
-  }
-
-  final Directory tempDir = await getTemporaryDirectory();
-  final String tempPath = tempDir.path;
-  final String filePath = '$tempPath/upload_img_file_01.tmp';
-  return File(filePath).writeAsBytes(bytes!);
-}
 
 Future<bool> uploadCapturedPicture(
   BuildContext context, {

--- a/packages/smooth_app/lib/l10n/app_en.arb
+++ b/packages/smooth_app/lib/l10n/app_en.arb
@@ -407,7 +407,9 @@
         "description": "Button clicking on which allows users to retake the last photo they took."
     },
     "uploading_image": "Uploading Image to the server",
-    "@uploading_image": {},
+    "@uploading_image": {
+        "description": "Message when a new picture is uploading to the server"
+    },
     "score_add_missing_ingredients": "Add missing ingredients",
     "score_add_missing_nutrition_facts": "Add missing nutrition facts",
     "score_add_missing_product_category": "Add missing product category",

--- a/packages/smooth_app/lib/pages/image_crop_page.dart
+++ b/packages/smooth_app/lib/pages/image_crop_page.dart
@@ -53,8 +53,9 @@ class ImageCropPage extends StatelessWidget {
               final Directory tempDir = await getTemporaryDirectory();
               final String tempPath = tempDir.path;
               final String filePath = '$tempPath/upload_img_file_01.tmp';
+              final File file = await File(filePath).writeAsBytes(image);
 
-              Navigator.pop(context, File(filePath).writeAsBytes(image));
+              Navigator.pop(context, file);
             },
             initialSize: 0.5,
             baseColor: theme.colorScheme.primary,

--- a/packages/smooth_app/lib/pages/image_crop_page.dart
+++ b/packages/smooth_app/lib/pages/image_crop_page.dart
@@ -31,59 +31,58 @@ class ImageCropPage extends StatelessWidget {
     final ThemeData theme = Theme.of(context);
     context.watch<ThemeProvider>();
 
-    return FutureBuilder<Uint8List?>(
-      future: pickImage(),
-      builder: (
-        BuildContext context,
-        AsyncSnapshot<Uint8List?> snap,
-      ) {
-        if (snap.connectionState != ConnectionState.done) {
-          return const Center(child: CircularProgressIndicator());
-        }
+    return Scaffold(
+        body: FutureBuilder<Uint8List?>(
+          future: pickImage(),
+          builder: (
+            BuildContext context,
+            AsyncSnapshot<Uint8List?> snap,
+          ) {
+            if (snap.connectionState != ConnectionState.done) {
+              return const Center(child: CircularProgressIndicator());
+            }
 
-        if (snap.data == null) {
-          Navigator.pop(context);
-        }
+            if (snap.data == null) {
+              Navigator.pop(context);
+            }
 
-        return Scaffold(
-          body: Crop(
-            image: snap.data!,
-            controller: _controller,
-            onCropped: (Uint8List image) async {
-              final Directory tempDir = await getTemporaryDirectory();
-              final String tempPath = tempDir.path;
-              final String filePath = '$tempPath/upload_img_file.tmp';
-              final File file = await File(filePath).writeAsBytes(image);
+            return Crop(
+              image: snap.data!,
+              controller: _controller,
+              onCropped: (Uint8List image) async {
+                final Directory tempDir = await getTemporaryDirectory();
+                final String tempPath = tempDir.path;
+                final String filePath = '$tempPath/upload_img_file.tmp';
+                final File file = await File(filePath).writeAsBytes(image);
 
-              Navigator.pop(context, file);
-            },
-            initialSize: 0.5,
-            baseColor: theme.colorScheme.primary,
-            maskColor: Colors.white.withAlpha(100),
-            cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
-                DotControl(color: theme.colorScheme.primary),
+                Navigator.pop(context, file);
+              },
+              initialSize: 0.5,
+              baseColor: theme.colorScheme.primary,
+              maskColor: Colors.white.withAlpha(100),
+              cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
+                  DotControl(color: theme.colorScheme.primary),
+            );
+          },
+        ),
+        bottomNavigationBar: BottomAppBar(
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: <Widget>[
+              IconButton(
+                icon: const Icon(Icons.cancel_outlined),
+                onPressed: () {
+                  Navigator.pop(context);
+                },
+              ),
+              IconButton(
+                icon: const Icon(Icons.done),
+                onPressed: () {
+                  _controller.crop();
+                },
+              )
+            ],
           ),
-          bottomNavigationBar: BottomAppBar(
-            child: Row(
-              mainAxisAlignment: MainAxisAlignment.spaceBetween,
-              children: <Widget>[
-                IconButton(
-                  icon: const Icon(Icons.cancel_outlined),
-                  onPressed: () {
-                    Navigator.pop(context);
-                  },
-                ),
-                IconButton(
-                  icon: const Icon(Icons.done),
-                  onPressed: () {
-                    _controller.crop();
-                  },
-                )
-              ],
-            ),
-          ),
-        );
-      },
-    );
+        ));
   }
 }

--- a/packages/smooth_app/lib/pages/image_crop_page.dart
+++ b/packages/smooth_app/lib/pages/image_crop_page.dart
@@ -52,7 +52,7 @@ class ImageCropPage extends StatelessWidget {
             onCropped: (Uint8List image) async {
               final Directory tempDir = await getTemporaryDirectory();
               final String tempPath = tempDir.path;
-              final String filePath = '$tempPath/upload_img_file_01.tmp';
+              final String filePath = '$tempPath/upload_img_file.tmp';
               final File file = await File(filePath).writeAsBytes(image);
 
               Navigator.pop(context, file);

--- a/packages/smooth_app/lib/pages/image_crop_page.dart
+++ b/packages/smooth_app/lib/pages/image_crop_page.dart
@@ -1,0 +1,88 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:crop_your_image/crop_your_image.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+
+class ImageCropPage extends StatelessWidget {
+  ImageCropPage({Key? key}) : super(key: key);
+
+  final CropController _controller = CropController();
+  final ImagePicker picker = ImagePicker();
+
+  Future<Uint8List?> pickImage() async {
+    final XFile? pickedXFile = await picker.pickImage(
+      source: ImageSource.camera,
+    );
+    if (pickedXFile == null) {
+      // User didn't pick any image.
+      return null;
+    }
+
+    return pickedXFile.readAsBytes();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    context.watch<ThemeProvider>();
+
+    return FutureBuilder<Uint8List?>(
+      future: pickImage(),
+      builder: (
+        BuildContext context,
+        AsyncSnapshot<Uint8List?> snap,
+      ) {
+        if (snap.connectionState != ConnectionState.done) {
+          return const Center(child: CircularProgressIndicator());
+        }
+
+        if (snap.data == null) {
+          Navigator.pop(context);
+        }
+
+        return Scaffold(
+          body: Crop(
+            image: snap.data!,
+            controller: _controller,
+            onCropped: (Uint8List image) async {
+              final Directory tempDir = await getTemporaryDirectory();
+              final String tempPath = tempDir.path;
+              final String filePath = '$tempPath/upload_img_file_01.tmp';
+
+              Navigator.pop(context, File(filePath).writeAsBytes(image));
+            },
+            initialSize: 0.5,
+            baseColor: theme.colorScheme.primary,
+            maskColor: Colors.white.withAlpha(100),
+            cornerDotBuilder: (double size, EdgeAlignment edgeAlignment) =>
+                DotControl(color: theme.colorScheme.primary),
+          ),
+          bottomNavigationBar: BottomAppBar(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                IconButton(
+                  icon: const Icon(Icons.cancel_outlined),
+                  onPressed: () {
+                    Navigator.pop(context);
+                  },
+                ),
+                IconButton(
+                  icon: const Icon(Icons.done),
+                  onPressed: () {
+                    _controller.crop();
+                  },
+                )
+              ],
+            ),
+          ),
+        );
+      },
+    );
+  }
+}

--- a/packages/smooth_app/lib/pages/product/add_new_product_page.dart
+++ b/packages/smooth_app/lib/pages/product/add_new_product_page.dart
@@ -6,8 +6,8 @@ import 'package:openfoodfacts/model/ProductImage.dart';
 import 'package:provider/provider.dart';
 import 'package:smooth_app/generic_lib/buttons/smooth_action_button.dart';
 import 'package:smooth_app/generic_lib/buttons/smooth_large_button_with_icon.dart';
-import 'package:smooth_app/helpers/picture_capture_helper.dart';
 import 'package:smooth_app/helpers/ui_helpers.dart';
+import 'package:smooth_app/pages/image_crop_page.dart';
 import 'package:smooth_app/pages/product/confirm_and_upload_picture.dart';
 import 'package:smooth_app/themes/theme_provider.dart';
 
@@ -123,7 +123,12 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
         icon: Icons.camera_alt,
         isDarkMode: themeProvider.darkTheme,
         onPressed: () async {
-          final File? initialPhoto = await pickImageAndCrop(context);
+          final File? initialPhoto = await Navigator.push<File?>(
+            context,
+            MaterialPageRoute<File?>(
+              builder: (BuildContext context) => ImageCropPage(),
+            ),
+          );
           if (initialPhoto == null) {
             return;
           }

--- a/packages/smooth_app/lib/pages/product/add_new_product_page.dart
+++ b/packages/smooth_app/lib/pages/product/add_new_product_page.dart
@@ -123,7 +123,7 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
         icon: Icons.camera_alt,
         isDarkMode: themeProvider.darkTheme,
         onPressed: () async {
-          final File? initialPhoto = await pickImageAndCrop();
+          final File? initialPhoto = await pickImageAndCrop(context);
           if (initialPhoto == null) {
             return;
           }

--- a/packages/smooth_app/lib/pages/product/add_new_product_page.dart
+++ b/packages/smooth_app/lib/pages/product/add_new_product_page.dart
@@ -149,6 +149,7 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
             _uploadedImages[imageType]!.add(initialPhoto);
             setState(() {});
           }
+          initialPhoto.delete();
         },
       ),
     );

--- a/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
+++ b/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
@@ -7,6 +7,8 @@ import 'package:smooth_app/generic_lib/buttons/smooth_action_button.dart';
 import 'package:smooth_app/helpers/picture_capture_helper.dart';
 import 'package:smooth_app/helpers/ui_helpers.dart';
 
+import '../image_crop_page.dart';
+
 class ConfirmAndUploadPicture extends StatefulWidget {
   const ConfirmAndUploadPicture({
     required this.barcode,
@@ -61,7 +63,13 @@ class _ConfirmAndUploadPictureState extends State<ConfirmAndUploadPicture> {
                         text: appLocalizations.retake_photo_button_label,
                         onPressed: () async {
                           final File? retakenPhoto =
-                              await pickImageAndCrop(context);
+                              await Navigator.push<File?>(
+                            context,
+                            MaterialPageRoute<File?>(
+                              builder: (BuildContext context) =>
+                                  ImageCropPage(),
+                            ),
+                          );
                           if (retakenPhoto == null) {
                             // User chose not to upload the image.
                             Navigator.pop(context);

--- a/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
+++ b/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
@@ -60,7 +60,8 @@ class _ConfirmAndUploadPictureState extends State<ConfirmAndUploadPicture> {
                     SmoothActionButton(
                         text: appLocalizations.retake_photo_button_label,
                         onPressed: () async {
-                          final File? retakenPhoto = await pickImageAndCrop();
+                          final File? retakenPhoto =
+                              await pickImageAndCrop(context);
                           if (retakenPhoto == null) {
                             // User chose not to upload the image.
                             Navigator.pop(context);

--- a/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
+++ b/packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart
@@ -78,6 +78,7 @@ class _ConfirmAndUploadPictureState extends State<ConfirmAndUploadPicture> {
                           setState(() {
                             photo = retakenPhoto;
                           });
+                          retakenPhoto.delete();
                         }),
                     SmoothActionButton(
                         text: _getConfirmButtonText(

--- a/packages/smooth_app/pubspec.yaml
+++ b/packages/smooth_app/pubspec.yaml
@@ -23,7 +23,6 @@ dependencies:
   hive: ^2.0.5
   hive_flutter: ^1.1.0
   http: ^0.13.4
-  image_cropper: ^1.4.1
   image_picker: ^0.8.4+4
   iso_countries: 2.1.0
   latlong2: ^0.8.1
@@ -44,6 +43,7 @@ dependencies:
   google_ml_barcode_scanner: ^0.0.2
   camera: ^0.9.4+5
   percent_indicator: ^4.0.0
+  crop_your_image: ^0.6.0+1
 
 
   

